java/java.editor.base/src/org/netbeans/modules/java/editor/base/semantic/SemanticHighlighterBase.java

```diff
@@ -573,6 +582,27 @@ private void addModifiers(Element decl, Collection<ColoringAttributes> c) {
             return null;
         }
 
+        private boolean isRecordComponent(ElementKind kind) {
+            return "RECORD_COMPONENT".equals(kind.name());
+        }
+
+        private Element toRecordComponent(Element el) {
```

Comment:
Can TreeShims.toRecordComponent be used here?

Reply:
Comment addressed.